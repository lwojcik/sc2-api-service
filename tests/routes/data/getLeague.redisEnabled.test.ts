import fastify from 'fastify';
import fastifyRedis from 'fastify-redis-mock';
import server from '../../../src/index';
import getConfig from '../../helper';

describe('/data/league/:seasonId/:queueId/:teamType/:leagueId (Redis enabled)', () => {
  const fastifyServer = fastify() as any;
  const url = '/data/league/40/201/0/6';
  const cacheSegment = 'league-40-201-0-6';
  const expectedTTL = getConfig(true).redis.ttl.league;

  beforeAll(() => {
    fastifyServer.register(fastifyRedis, {
      host: '127.0.0.1',
      port: '6379',
      password: '',
      enableReadyCheck: true,
      dropBufferSupport: false,
    });
    fastifyServer.register(server, getConfig(true));
  });

  afterAll(() => fastifyServer.close());

  it('returns 200', async () => {
    const res = await fastifyServer.inject({ method: 'GET', url });
    expect(res.statusCode).toBe(200);
  });

  it('returns correct response', async () => {
    const res = await fastifyServer.inject({ method: 'GET', url });
    expect(res.payload).toMatchSnapshot();
  });

  it('response is cached correctly', async () => {
    const cachedResponse = await fastifyServer.redis.get(cacheSegment);
    expect(cachedResponse).toMatchSnapshot();
  });

  it('cached response has correct TTL', async () => {
    const ttl = await fastifyServer.redis.ttl(cacheSegment);
    expect(ttl).toEqual(expectedTTL);
  });

  it('returns correct response when refresh is set to true', async () => {
    const res = await fastifyServer.inject({ method: 'GET', url, query: { refresh: 'true' } });
    expect(res.statusCode).toBe(200);
    expect(res.payload).toMatchSnapshot();
  });
});