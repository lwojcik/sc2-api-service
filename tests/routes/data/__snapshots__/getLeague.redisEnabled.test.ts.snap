// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/data/league/:seasonId/:queueId/:teamType/:leagueId (Redis enabled) response is cached correctly 1`] = `"{\\"status\\":200,\\"data\\":{\\"_links\\":{\\"self\\":{\\"href\\":\\"sample league object\\"}}}}"`;

exports[`/data/league/:seasonId/:queueId/:teamType/:leagueId (Redis enabled) returns correct response 1`] = `"{\\"status\\":200,\\"data\\":{\\"_links\\":{\\"self\\":{\\"href\\":\\"sample league object\\"}}}}"`;

exports[`/data/league/:seasonId/:queueId/:teamType/:leagueId (Redis enabled) returns correct response when refresh is set to true 1`] = `"{\\"status\\":200,\\"data\\":{\\"_links\\":{\\"self\\":{\\"href\\":\\"sample league object\\"}}}}"`;
